gsi_openssh/source/fips_mode_replacement.h

```diff
@@ -14,12 +14,16 @@
  * limitations under the License.
  */
 
+#ifndef FIPS_MODE_REPLACEMENT_H
+#define FIPS_MODE_REPLACEMENT_H
+
 #if OPENSSL_VERSION_NUMBER >= 0x30000000L
 /*
- * OpenSSL version 3.0 and up no longer has FIPS_mode().
- * Making a replacement function is not feasible since FIPS would need to be
- * initialized differently in any case.
- * See https://www.openssl.org/docs/manmaster/man7/fips_module.html for details
+ * OpenSSL versions 3.0 and up no longer have FIPS_mode(). To support both
+ * OpenSSL 3.x and older versions for other OSes, we use the replacement
+ * function as shipped by Fedora/RHEL/CentOS in their OpenSSL 3.x packages.
  */
-# define FIPS_mode() 0
-#endif
\ No newline at end of file
+# define FIPS_mode() EVP_default_properties_is_fips_enabled(NULL)
```

Comment:
but now you're still missing the protection for (re)defining FIPS_mode().
I think the whole file should be - code only - :
```
#ifndef FIPS_MODE_REPLACEMENT_H
#define FIPS_MODE_REPLACEMENT_H

#if OPENSSL_VERSION_NUMBER >= 0x30000000L
# ifndef FIPS_mode
#  define FIPS_mode() EVP_default_properties_is_fips_enabled(NULL)
# endif /* FIPS_mode */
#endif /* openssl */

#endif /* FIPS_MODE_REPLACEMENT_H */
```

Reply:
Strange, don't know why I missed that, as I wrote about it just prior to making the change. Well, it was done in a hurry, maybe that's the reason. Should be correct now.